src/main/java/frc/robot/subsystems/cellmech/Hopper.java

```diff
@@ -27,4 +100,304 @@ public static Hopper getInstance() {
 
     }
 
+    @Override
+    public void periodic() {
+
+        // Count cells
+
+        // cache values of line break sensors
+        boolean bottomValue = m_lineBottom.get();
+        boolean topValue = m_lineTop.get();
+
+        // If belt is moving up
+        if (m_hopperBelt.get() > 0.0) {
+
+            // add when cell enters bottom
+            if (bottomValue == true && m_lineBottomLastValue == false) {
+                m_cellCount += 1;
+            }
+
+            // subtract when cell exits top
+            if (topValue == false && m_lineTopLastValue == true) {
+                m_cellCount -= 1;
+            }
+
+            // If belt is moving down
+        } else {
+
+            // subtract when cell exits bottom
+            if (bottomValue == false && m_lineBottomLastValue == true) {
+                m_cellCount -= 1;
+            }
+
+            // add when cell enters top
+            if (topValue == true && m_lineTopLastValue == false) {
+                m_cellCount += 1;
+            }
```

Comment:
Holding a ball at any of these sensors will cause the ball count to rapidly increase or drop. These should be latched